app/src/shared/web3/web3Poller/web3Poller.js

```diff
@@ -0,0 +1,185 @@
+    ACCOUNT_ERROR: 'WEB3POLLER/ACCOUNT_ERROR',
+    NETWORK: 'WEB3POLLER/NETWORK',
+    NETWORK_ERROR: 'WEB3POLLER/NETWORK_ERROR',
+    TRANSACTION_COMPLETE: 'WEB3POLLER/TRANSACTION_COMPLETE',
+    TRANSACTION_ERROR: 'WEB3POLLER/TRANSACTION_ERROR',
+}
+
+export type Event = $Values<typeof events>
+export type Handler = (any, any) => void
+
+const WEB3_POLL_INTERVAL = 1000 // 1s
+const NETWORK_POLL_INTERVAL = 1000 // 1s
+const PENDING_TX_POLL_INTERVAL = 1000 * 5 // 5s
+
+export default class Web3Poller {
+    web3PollTimeout: ?TimeoutID = null
+    ethereumNetworkPollTimeout: ?TimeoutID = null
+    pendingTransactionsPollTimeout: ?TimeoutID = null
+    web3: StreamrWeb3Type = getWeb3()
+    account: any = null
+    networkId: NumberString = ''
+    emitter: EventEmitter = new EventEmitter()
+
+    constructor() {
+        // Start polling for info
+        this.pollWeb3()
+        this.pollEthereumNetwork()
+        this.pollPendingTransactions()
+    }
+
+    subscribe(event: Event, handler: Handler) {
+        this.emitter.on(event, handler)
+    }
+
+    unsubscribe(event: Event, handler: Handler) {
+        this.emitter.removeListener(event, handler)
+    }
+
+    startWeb3Poll = () => {
+        this.clearWeb3Poll()
+        this.web3PollTimeout = setTimeout(this.pollWeb3, WEB3_POLL_INTERVAL)
+    }
+
+    pollWeb3 = () => (
+        this.fetchWeb3Account()
+            .then(this.startWeb3Poll, this.startWeb3Poll)
+    )
+
+    startEthereumNetworkPoll = () => {
+        this.clearEthereumNetworkPoll()
+        this.ethereumNetworkPollTimeout = setTimeout(this.pollEthereumNetwork, NETWORK_POLL_INTERVAL)
+    }
+
+    pollEthereumNetwork = () => (
+        this.fetchChosenEthereumNetwork()
+            .then(this.startEthereumNetworkPoll, this.startEthereumNetworkPoll)
+    )
+
+    clearWeb3Poll = () => {
+        if (this.web3PollTimeout) {
+            clearTimeout(this.web3PollTimeout)
+            this.web3PollTimeout = null
+        }
+    }
+
+    clearEthereumNetworkPoll = () => {
+        if (this.ethereumNetworkPollTimeout) {
+            clearTimeout(this.ethereumNetworkPollTimeout)
+            this.ethereumNetworkPollTimeout = null
+        }
+    }
+
+    clearPendingTransactionsPoll = () => {
+        if (this.pendingTransactionsPollTimeout) {
+            clearTimeout(this.pendingTransactionsPollTimeout)
+            this.pendingTransactionsPollTimeout = null
+        }
+    }
+
+    fetchWeb3Account = () => (
+        this.web3.getDefaultAccount()
+            .then((account) => {
+                this.handleAccount(account)
+                // needed to avoid warnings about creating promise inside a handler
+                // if any other web3 actions are dispatched.
+                return Promise.resolve()
+            }, (err) => {
+                if (this.account) {
+                    this.account = null
+                    this.emitter.emit(events.ACCOUNT_ERROR, err)
+                }
```

Comment:
Failing here is ok since this method is called every second and the error is emitted only when a user was signed in already. Otherwise we will be spamming the error to console.